### PR TITLE
Artisanal Production Builds By Chad

### PR DESCRIPTION
**Pull request by chadhietala**
So some how this has caused 54 test failures, where I think majority of them are serializing an object into the DOM. Unsure why that is since `expect`, `assert`, and `unwrap` were basically just pass throughs.
screen shot 2017-03-25 at 8 20 20 pm

### CODE REVIEW

File: packages/@glimmer/runtime/lib/compiled/opcodes/dom.ts [hunk 1]
```diff
@@ -417,7 +410,7 @@ export class DynamicAttribute implements Attribute  {
   patch(env: Environment) {
     let { element, cache } = this;
 
-    let value = expect(cache, 'must patch after flush').revalidate();
+    let value = cache;
```
Comment (GavinJoyce) [inline packages/@glimmer/runtime/lib/compiled/opcodes/dom.ts]: this should be `let value = cache.revalidate();`

Comment (chadhietala) [inline packages/@glimmer/runtime/lib/compiled/opcodes/dom.ts]: good 👁

**Comment by chadhietala** [comment]
screen shot 2017-03-26 at 9 05 36 am

**Comment by GavinJoyce** [comment]
@tomdale I'm getting the following error after a `yarn upgrade`:

```
File: src/ui/components/my-table/template.hbs
The Broccoli Plugin: [broccoli-persistent-filter:GlimmerTemplatePrecompiler] failed with:
TypeError: (0 , _util.expect) is not a function
    at TemplateVisitor.get (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/template-visitor.js:390:37)
    at TemplateVisitor.TextNode (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/template-visitor.js:331:25)
    at TemplateVisitor.visit (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/template-visitor.js:248:24)
    at TemplateVisitor.Program (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/template-visitor.js:281:18)
    at TemplateVisitor.visit (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/template-visitor.js:248:24)
    at Function.compile (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/template-compiler.js:55:25)
    at Object.precompile (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/@glimmer/compiler/dist/commonjs/es5/lib/compiler.js:49:60)
    at GlimmerTemplatePrecompiler.processString (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/@glimmer/application-pipeline/dist/broccoli/glimmer-template-precompiler.js:14:45)
    at /Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/broccoli-persistent-filter/lib/strategies/default.js:10:19
    at initializePromise (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/rsvp/dist/rsvp.js:589:5)

The broccoli plugin was instantiated at:
    at GlimmerTemplatePrecompiler.Plugin (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/broccoli-persistent-filter/node_modules/broccoli-plugin/index.js:7:31)
    at GlimmerTemplatePrecompiler.Filter [as constructor] (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/broccoli-persistent-filter/index.js:62:10)
    at GlimmerTemplatePrecompiler (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/@glimmer/application-pipeline/dist/broccoli/glimmer-template-precompiler.js:7:9)
    at GlimmerApp.compiledHandlebarsTree (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/@glimmer/application-pipeline/dist/broccoli/glimmer-app.js:184:16)
    at GlimmerApp.javascriptTree (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/@glimmer/application-pipeline/dist/broccoli/glimmer-app.js:157:45)
    at GlimmerApp.toTree (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/@glimmer/application-pipeline/dist/broccoli/glimmer-app.js:127:27)
    at module.exports (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/ember-cli-build.js:21:14)
    at Class.setupBroccoliBuilder (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/ember-cli/lib/models/builder.js:72:19)
    at Class.init (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/ember-cli/lib/models/builder.js:52:10)
    at Class.superWrapper [as init] (/Users/gavin/dev/opensource/glimmerbench/preact-glimmer/node_modules/core-object/lib/assign-properties.js:32:18)
```

**Comment by GavinJoyce** [comment]
I've got `@glimmer/compiler 0.23.0-alpha.7` but it seems to be missing Chad's changes:

screen shot 2017-03-26 at 09 54 28

screen shot 2017-03-26 at 09 53 33

@tomdale perhaps there is a bug in the release script or you need to `git pull`?

**Pull request merged**